site-src/geps/gep-718.md

```diff
@@ -0,0 +1,223 @@
+	// The controller should raise the "ResolvedRefs" condition on the Gateway
+	// with the "DegradedRoutes" reason. The gateway status for this route should
+	// be updated with a condition that describes the error more specifically.
+	//
+	// The protocol to use is defined using AppProtocol field (introduced in
+	// Kubernetes 1.18) in the Service resource. In the absence of the
+	// AppProtocol field a `gateway.networking.k8s.io/app-protocol` annotation on the
+	// BackendPolicy resource may be used to define the protocol. If the
+	// AppProtocol field is available, this annotation should not be used. The
+	// AppProtocol field, when populated, takes precedence over the annotation
+	// in the BackendPolicy resource. For custom backends, it is encouraged to
+	// add a semantically-equivalent field in the Custom Resource Definition.
+	//
+	// Support: Core (Kubernetes Service)
+	// Support: Custom (any other resource)
+	//
+	// +optional
+    // +kubebuilder:default={kind: "service"}
+	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
+
+	// Port specifies the destination port number to use for the
+	// backend referenced by BackendRef field. Port is required when the backend
+    // is a Kubernetes Service.
+	//
+	// Support: Core
+	//
+	// +optional
+	Port *PortNumber `json:"port,omitempty"`
+
+	// Weight specifies the proportion of HTTP requests forwarded to the backend
+	// referenced by the ServiceName or BackendRef field. This is computed as
+	// weight/(sum of all weights in this ForwardTo list). For non-zero values,
+	// there may be some epsilon from the exact proportion defined here
+	// depending on the precision an implementation supports. Weight is not a
+	// percentage and the sum of weights does not need to equal 100.
+	//
+	// If only one backend is specified and it has a weight greater than 0, 100%
+	// of the traffic is forwarded to that backend. If weight is set to 0, no
+	// traffic should be forwarded for this entry. If unspecified, weight
+	// defaults to 1.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1000000
+	Weight *int32 `json:"weight,omitempty"`
+}
+
+```
+
+For Kubernetes Services, the API will look like the following:
+
+```yaml
+...
+forwardTo:
+- backendRef:
+    name: foo-service-v1
+  port: 80
+  weight: 80
+- backendRef:
+    name: foo-service-canary
+  port: 80
+  weight: 20
+...
+```
+
+Here, the `kind` field is omitted as it will be injected as a default.
+
+For custom backends, the API will look like the following:
+
+```yaml
+...
+forwardTo:
+- backendRef:
+    name: foo-v1
+    kind: server
+    group: networking.acme.io
+  port: 80
+  weight: 80
+- backendRef:
+    name: foo-v1-canary
+    kind: server
+    group: networking.acme.io
+  port: 80
+  weight: 20
+...
+```
+
+## Note on future consolidation
+
+It has been noted that `forwardTo` doesn't flow as good as `backendRefs`.
```

Comment:
Question - why wait to make this consolidation? Every change will be harder in the future and v1alpha2 is the best milestone to make these changes. The `forwardTo` block has a big impact on the YAML size and new user experience. Even the most basic Route with no matches must use it.

Reply:
I should have used the word `future` more carefully.
I'll be writing another small GEP to dedicate a discussion on this matter and then implement this and the future GEP together for v1a2.

Reply:
My preference is the structure shown here. I read it as "forward to target X, with parameters A,B,C". I don't think that collapsing improves the readability.